### ISSUE TITLE
New Error After Update: module 'torch' has no attribute 'float8_e4m3fnuz'

### ISSUE THREAD

**Issue by Orenji-Tangerine** [kijai/ComfyUI-FluxTrainer#47]
![image](https://github.com/user-attachments/assets/1124e5be-217d-4385-a6f2-cda12a06e20b)

My pytorch version is 2.1.2+ cu118,  only workable after reverting back to commit [77587e0](https://github.com/kijai/ComfyUI-FluxTrainer/commit/77587e019315ceeae0b83c34fc71c2cc4ed9e4d6)

![image](https://github.com/user-attachments/assets/d12cd404-7dc1-4f54-a850-c4bec21930f2)

**Comment by Bellzs** [comment]
have you update the torch version？does it work？I am using the same torch version:2.1.2+cu118

**Comment by Orenji-Tangerine** [comment]
> have you update the torch version？does it work？I am using the same torch version:2.1.2+cu118

I have separate environments that run different pytorch versions ((like 2.3.1+ cu11.8, 2.3.1 + cu12.1) and it seems that only 2.1.2+cu11.8 (for anything below 2.4.0) works best without error for commit 77587e0 and below. Pytorch version before 2.4.0 will be OOM on my system. I havent tried 2.4.0 yet. xformers 0.0.28 post1 is out today to match pytorch 2.4.1 + cu12.4, I will try someday for this

**Comment by Orenji-Tangerine** [comment]
I have updated the environment
Pytorch 2.4.1+ cu12.4 with Python 3.11.6
So long the gradient checkpointing enabled with cpu offloading, there will be no OOM issues.
I did notice iteration speed increased from 4.xx s/it to 5.2-.5.3 s/it (training under 512*512 with CAME optimizer, memory peak around 14.3GB). But the total expected generation time does not differ too much
(PC running with 4060Ti 16GB + 64GB DDR4)
